saleor/graphql/discount/types/sales.py

```diff
@@ -121,81 +112,101 @@ class Meta:
             "and are visible to all the customers."
         )
         interfaces = [relay.Node, ObjectWithMetadata]
-        model = models.Sale
+        model = models.Promotion
+        doc_category = DOC_CATEGORY_DISCOUNTS
+
+    @staticmethod
+    def resolve_created(root: ChannelContext[models.Promotion], _info: ResolveInfo):
+        return root.node.created_at
 
     @staticmethod
-    def resolve_created(root: models.Sale, _info: ResolveInfo):
-        return root.created_at
+    def resolve_type(root: ChannelContext[models.Promotion], _info: ResolveInfo):
+        if rule := root.node.rules.first():
+            return rule.reward_value_type
 
     @staticmethod
     def resolve_categories(
-        root: ChannelContext[models.Sale], info: ResolveInfo, **kwargs
+        root: ChannelContext[models.Promotion], info: ResolveInfo, **kwargs
     ):
-        qs = root.node.categories.all()
-        return create_connection_slice(qs, info, kwargs, CategoryCountableConnection)
+        from ..utils import convert_migrated_sale_catalogue_predicate
+
+        if predicates := convert_migrated_sale_catalogue_predicate(root.node):
+            if category_ids := predicates.get("categoryPredicate"):
+                qs = Category.objects.filter(id__in=category_ids)
+                return create_connection_slice(
+                    qs, info, kwargs, CategoryCountableConnection
+                )
 
     @staticmethod
-    def resolve_channel_listings(root: ChannelContext[models.Sale], info: ResolveInfo):
-        return SaleChannelListingBySaleIdLoader(info.context).load(root.node.id)
+    def resolve_channel_listings(
+        root: ChannelContext[models.Promotion], info: ResolveInfo
+    ):
+        if rule := root.node.rules.first():
+            if channel := rule.channels.first():
+                return [
+                    SaleChannelListing(
+                        # TODO what about id???
+                        id=graphene.Node.to_global_id("SaleChannelListing", rule.id),
+                        channel=channel,
+                        discount_value=rule.reward_value,
+                        currency=channel.currency_code,
+                    )
```

Comment:
We will not have `PromotionChannelListing`, so it is kind of imitation, but what can we return here as `id` to not have a breaking change?